pkg/controllers/managementuser/snapshotbackpopulate/snapshotbackpopulate.go

```diff
@@ -71,13 +86,21 @@ func (h *handler) OnChange(key string, configMap *corev1.ConfigMap) (runtime.Obj
 		return configMap, nil
 	}
 
+	if h.activeConfigMap == "" {
+		h.activeConfigMap = configMap.Name
+	}
+
 	clusters, err := h.clusterCache.GetByIndex(cluster2.ByCluster, h.clusterName)
 	if err != nil || len(clusters) != 1 {
 		return configMap, fmt.Errorf("error while retrieving cluster %s from cache via index: %w", h.clusterName, err)
 	}
 
 	cluster := clusters[0]
 
+	if h.v1ClusterName == "" {
+		h.v1ClusterName = cluster.Name
+	}
+
```

Comment:
Can this be put in the `Register` function by getting the v1 cluster with the index `ByCluster`?

Reply:
The indexers/caches may not be synced when `Register` is called. We can artificially slow the controller down from running until those caches are synced, but I think it's fine to just leave this doing what it is doing now.